src/game/AI/ScriptDevAI/scripts/outland/hellfire_citadel/shattered_halls/shattered_hallsScripts.cpp

```diff
@@ -0,0 +1,279 @@
+                    SendAIEvent(AI_EVENT_CUSTOM_A, m_creature, partner);
+                    m_creature->getThreatManager().modifyThreatPercent(partner, -101);
+                    m_creature->ClearTemporaryFaction();
+                    m_eventStarted = false;
+                }
+            }
+        }
+    }
+};
+
+struct npc_shattered_hand_centurion : public CombatAI
+{
+    npc_shattered_hand_centurion(Creature* creature) : CombatAI(creature, GLADIATOR_ACTION_MAX),
+        m_instance(static_cast<ScriptedInstance*>(creature->GetInstanceData())),
+        m_eventStarted(false)
+    {
+        AddCustomAction(GLADIATOR_START_EVENT, 4000u, [&]() { HandleEventStart(); });
+    }
+
+    void Reset() override
+    {
+        CombatAI::Reset();
+        m_eventStarted = false;
+    }
+
+    ScriptedInstance* m_instance;
+    bool m_eventStarted;
+
+    void Aggro(Unit* /*who*/) override
+    {
+        switch (urand(0, 5))
+        {
+        case 0: DoBroadcastText(SAY_AGGRO1, m_creature); break;
+        case 1: DoBroadcastText(SAY_AGGRO2, m_creature); break;
+        case 2: DoBroadcastText(SAY_AGGRO3, m_creature); break;
+        case 3: DoBroadcastText(SAY_AGGRO4, m_creature); break;
+        case 4: DoBroadcastText(SAY_AGGRO5, m_creature); break;
+        case 5: DoBroadcastText(SAY_AGGRO6, m_creature); break;
+        case 6: DoBroadcastText(SAY_AGGRO7, m_creature); break;
+        }
+
+        // Set Gladiators infight too and reset their factions
+        CreatureList gladiatorList;
+        GetCreatureListWithEntryInGrid(gladiatorList, m_creature, NPC_SHATTERED_HAND_GLADIATIOR, 20.0f);
+        for (Creature* gladiator : gladiatorList)
+        {
+            if (gladiator->IsAlive())
+                SendAIEvent(AI_EVENT_CUSTOM_A, m_creature, gladiator);
+        }
+    }
+
+    void JustReachedHome() override
+    {
+        CombatAI::JustReachedHome();
+    }
+
+    void HandleEventStart()
+    {
+        if (m_creature->IsInCombat())
+            return;
+
+        if (m_eventStarted)
+            return;
+
+        // Get all 4 Gladiators in Range 
+        std::list<Creature*> m_gladiator;
+        GuidVector m_gladiatorGuid;
+        GetCreatureListWithEntryInGrid(m_gladiator, m_creature, NPC_SHATTERED_HAND_GLADIATIOR, 10.0f);
+        for (Creature* gladiator : m_gladiator)
+            m_gladiatorGuid.push_back(gladiator->GetObjectGuid());
+
+        // There should always be 4 Gladiators up handled via spawn_group flag respawn together
+        if (m_gladiatorGuid.size() != 4)
+            return;
+
+        // 2 of them get faction 1693
+        // the other 2 get faction 1692
+        Creature* first = m_creature->GetMap()->GetCreature(m_gladiatorGuid[0]);  // A  
+        Creature* second = m_creature->GetMap()->GetCreature(m_gladiatorGuid[1]); // B
+        Creature* third = m_creature->GetMap()->GetCreature(m_gladiatorGuid[2]); // A
+        Creature* forth = m_creature->GetMap()->GetCreature(m_gladiatorGuid[3]); // B
```

Comment:
I appreciated how efficiently you solved this with just 4 pointers.
I would've done something stupid and pointless like
std::tuple<Creature*, Creature*> gladiatorPairA(m_creature->GetMap()->GetCreature(m_gladiatorGuid[0], m_creature->GetMap()->GetCreature(m_gladiatorGuid[1]))
or something :laughing: